pkg/webui/console/components/mac-settings-section/index.js

```diff
@@ -469,43 +465,71 @@ const MacSettingsSection = props => {
         />
       </Form.FieldContainer>
       <Form.Field
-        name="mac_settings.use_adr"
-        component={Checkbox}
-        label={m.useAdr}
-        onChange={handleUseAdrChange}
+        name="mac_settings.adr"
+        component={Radio.Group}
+        title={m.adrAdaptiveDataRate}
         tooltipId={tooltipIds.ADR_USE}
-      />
-      {useAdr && (
+        encode={encodeAdrMode}
+        decode={decodeAdrMode}
+      >
+        <Radio label={m.adrDynamic} value="dynamic" />
+        <Radio label={m.adrStatic} value="static" />
+        <Radio label={m.adrDisabled} value="disabled" />
+      </Form.Field>
+      {isDynamicAdr && (
+        <Form.Field
+          title={m.adrMargin}
+          name="mac_settings.adr.dynamic.margin"
+          component={Input}
+          type="number"
+          tooltipId={tooltipIds.ADR_MARGIN}
+          inputWidth="xs"
+          append="dB"
```

Comment:
Any min/max we can use here @adriansmares ?

Reply:
No, but having something like [-100, 100] as a Console sanity check is fine. The backend endures any value.

In general the indices (the settings in static and dynamic) can now be validated using https://github.com/TheThingsNetwork/lorawan-stack/issues/5403 (you can even render only valid options if you want).